app/src/main/java/com/fstyle/structure_android/utils/Validator.java

```diff
@@ -0,0 +1,293 @@
+            }
+        }
+    }
+
+    @StringRes
+    public int getErrorByTypeInt(@ValidType int type) {
+        int index = mErrorTraces.indexOfValue(type);
+        if (index == -1) {
+            return R.string.empty;
+        }
+        return mErrorTraces.keyAt(index);
+    }
+
+    public String getErrorByType(@ValidType int type) {
+        int index = mErrorTraces.indexOfValue(type);
+        if (index == -1 || mErrorTraces.keyAt(index) == -1) {
+            return null;
+        }
+        return mContext.getString(mErrorTraces.keyAt(index));
+    }
+
+    public void initNGWordPattern() {
+        Observable.create(new Observable.OnSubscribe<Pattern>() {
+            @Override
+            public void call(Subscriber<? super Pattern> subscriber) {
+                try {
+                    BufferedReader reader = new BufferedReader(
+                            new InputStreamReader(mContext.getAssets().open("ng-word")));
+                    StringBuffer buffer = new StringBuffer();
+                    String line;
+                    while ((line = reader.readLine()) != null) {
+                        if (buffer.length() == 0) {
+                            buffer.append("(").append(line.toLowerCase(Locale.ENGLISH));
+                        } else {
+                            buffer.append("|").append(line.toLowerCase(Locale.ENGLISH));
+                        }
+                    }
+                    subscriber.onNext(Pattern.compile(buffer.append(")").toString()));
+                } catch (IOException e) {
+                    subscriber.onError(e);
+                }
+            }
+        })
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new Action1<Pattern>() {
+                    @Override
+                    public void call(Pattern pattern) {
+                        mNGWordPattern = pattern;
+                    }
+                }, new Action1<Throwable>() {
+                    @Override
+                    public void call(Throwable throwable) {
+                        throw new ValidationException("Error when open ng-word", throwable);
+                    }
+                });
+    }
+
+    @ValidMethod(type = { ValidType.NG_WORD })
+    public boolean validateNGWord(String str) {
+        if (mNGWordPattern == null) {
+            throw new ValidationException(
+                    "NGWordPattern is null!!! \n Call initNGWordPattern() before call this method!",
+                    new NullPointerException());
+        }
+        return TextUtils.isEmpty(str) || !mNGWordPattern.matcher(str.toLowerCase(Locale.ENGLISH))
+                .find();
+    }
+
+    @Documented
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ ElementType.FIELD, ElementType.METHOD })
+    public @interface Optional {
+
+    }
+
+    @IntDef({ ValidType.NG_WORD })
+    public @interface ValidType {
+        int NG_WORD = 0x00;
```

Comment:
ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.